amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/polling/SynchronousBlockingRetrievalFactory.java

```diff
@@ -63,7 +63,7 @@ public SynchronousBlockingRetrievalFactory(String streamName, KinesisAsyncClient
     public GetRecordsRetrievalStrategy createGetRecordsRetrievalStrategy(@NonNull final ShardInfo shardInfo,
             @NonNull final MetricsFactory metricsFactory) {
         return new SynchronousGetRecordsRetrievalStrategy(
-                new KinesisDataFetcher(kinesisClient, streamName, shardInfo.shardId(), maxRecords, metricsFactory, kinesisRequestTimeout));
+                new KinesisDataFetcher(kinesisClient, shardInfo.streamName().orElse(streamName), shardInfo.shardId(), maxRecords, metricsFactory, kinesisRequestTimeout));
```

Comment:
doesn't shardInfo always have streamName?

Reply:
The current version of ShardInfo doesn't have streamName, as we do not have this this in DDB lease record, from which this is constructed.